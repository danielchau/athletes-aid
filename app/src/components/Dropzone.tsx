import React, { useCallback } from "react";
import { useDropzone } from "react-dropzone";
import { Typography } from "@material-ui/core";
import CloudUploadIcon from "@material-ui/icons/CloudUpload";
import { ExcelToJSON } from "../util/excelUpload";

interface MyDropzoneProps {
    setNewAthletes: any;
}

/**
 * Dropzone is an area that files can be uploaded to.
 * File has to be of the format of the one that is available for download in the previous component.
 * @param props
 */
export default function MyDropzone(props: MyDropzoneProps) {
    /**
     * Transforms a file from the drag and drop to an array of athletesProfiles.
     */
    const onDrop = useCallback(acceptedFiles => {
        acceptedFiles.forEach((file: any) => {
            const reader = new FileReader();

            reader.onabort = () => console.log("file reading was aborted");
            reader.onerror = () => console.log("file reading has failed");
            reader.onload = () => {
                const binaryStr = reader.result;
                let jsonObject = ExcelToJSON(binaryStr);
                let hasError = false;
                let athletes = jsonObject.map((entry: any, i: number) => {
                    if (
                        !!!entry["Athlete First Name*"] ||
                        !!!entry["Athlete Last Name*"] ||
                        !!!entry["Gender*"] ||
                        !!!entry["E-mail*"]
                    ) {
                        hasError = true;
                    }
                    return {
                        id: i,
                        profilePicture: "",
                        name: entry["Athlete First Name*"] + " " + entry["Athlete Last Name*"],
                        birthdate: new Date(entry["Day Of Birthday yyyy-mm-dd"]).toDateString(),
                        schoolYear: entry["School Year"],
                        gender: entry["Gender*"],
                        weight: entry["Weight in Lbs"],
                        height: entry["Height in Inches"],
                        email: entry["E-mail*"],
                        cellPhone: entry["Cell Phone"],
                        homePhone: entry["Home Phone"],
                        healthCardNumber: "",
                        emergencyContact: {
                            id: "e" + i,
                            name: entry["Emergency Contact Name"],
                            phone: entry["Emergency Contact Phone"],
                            email: entry["Emergency Contact Email"]
                        },
                        files: [],
                        injuries: []
                    };
                });
                if (hasError) {
                    alert("Please fill out all the required columns in the spreadsheet.");
                } else {
                    props.setNewAthletes(athletes);
                }
            };
            reader.readAsBinaryString(file);
        });
    }, []);
    const { getRootProps, getInputProps } = useDropzone({ onDrop });

    return (
        <div
            style={{
                height: "calc(100% - 16px)",
                display: "flex",
                flexGrow: 1,
                textAlign: "center",
                justifyContent: "center",
                alignContent: "center",
                alignItems: "center",
                border: "1px dashed #F2A71E",
                margin: "8px",
                flexDirection: "column"
            }}
            {...getRootProps()}
        >
            <input {...getInputProps()} />
            <CloudUploadIcon style={{ width: "2em", height: "2em" }}></CloudUploadIcon>
            <Typography>Drag 'n' drop some files here, or click to select files</Typography>
        </div>
    );
}
