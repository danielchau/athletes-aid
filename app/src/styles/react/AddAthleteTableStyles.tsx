import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

/**
 * React styles for the Add Athletes Table.
 */
export const addAthleteTableStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            padding: "8px",
            height: "100%",
            display: "flex",
            justifyContent: "center"
        },
        tableContainer: {
            width: "100%",
            height: "100%",
            padding: "8px",
            backgroundColor: "rgba(0, 33, 69, 0.05)",
            overflow: "scroll"
        },
        table: {
            width: "100%"
        },
        tableBodyContainer: {
            width: "2000px",
            overflow: "auto",
            height: "100%"
        },
        tableBody: {
            tableLayout: "fixed"
        },
        tableRow: {
            cursor: "pointer"
        },
        tooltip: {
            backgroundColor: "#fff",
            color: "#4f4f4f",
            boxShadow: theme.shadows[1]
        }
    })
);
