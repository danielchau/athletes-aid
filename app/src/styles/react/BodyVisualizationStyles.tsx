import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

/**
 * React styles for the Body Visualization.
 */
export const bodyVisualizationStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            height: "100%"
        },
        imageContainer: {
            position: "relative",
            alignSelf: "center"
        },
        image: {
            width: "100%"
        },
        dot: {
            height: "15px",
            width: "15px",
            position: "absolute",
            borderRadius: "15px"
        },
        legendContainer: {
            borderLeft: "#dedede solid 1px"
        },
        legendItem: {
            display: "flex",
            color: "rgba(0, 0, 0, 0.54)",
            marginBottom: "2px"
        },
        dotLegend: {
            height: "10px",
            width: "10px",
            borderRadius: "10px",
            marginTop: "6px",
            marginRight: "4px"
        }
    })
);
