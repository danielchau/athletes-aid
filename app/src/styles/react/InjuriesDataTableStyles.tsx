import { makeStyles, Theme, createStyles, lighten } from "@material-ui/core/styles";

/**
 * React styles for the Injury Data Table.
 */
export const injuriesDataTableStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            marginTop: theme.spacing(3)
        },
        paper: {
            width: "100%",
            marginBottom: theme.spacing(2),
            backgroundColor: "rgba(0, 33, 69, 0.05)"
        },
        table: {
            minWidth: 750
        },
        tableWrapper: {
            overflowX: "auto"
        },
        tableRow: {
            cursor: "pointer"
        },
        visuallyHidden: {
            border: 0,
            clip: "rect(0 0 0 0)",
            height: 1,
            margin: -1,
            overflow: "hidden",
            padding: 0,
            position: "absolute",
            top: 20,
            width: 1
        }
    })
);

/**
 * React styles for the Injury Data Table Toolbar Header.
 */
export const injuriesDataTableToolbarStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            paddingLeft: theme.spacing(2),
            paddingRight: theme.spacing(1)
        },
        highlight:
            theme.palette.type === "light"
                ? {
                      color: theme.palette.primary.main,
                      backgroundColor: lighten(theme.palette.primary.light, 0.85)
                  }
                : {
                      color: theme.palette.text.primary,
                      backgroundColor: theme.palette.secondary.dark
                  },
        title: {
            flex: "1 1 100%"
        },
        export: {
            margin: "auto",
            color: theme.palette.secondary.main,
            display: "flex"
        }
    })
);
