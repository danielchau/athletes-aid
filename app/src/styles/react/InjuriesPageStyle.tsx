import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

/**
 * React styles for the Injuries Page.
 */
export const injuriesPageStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            padding: "16px",
            backgroundColor: "#fff"
        },
        drawerOpen: {
            width: "calc(100vw - 240px)",
            transition: theme.transitions.create("width", {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.enteringScreen
            }),
            [theme.breakpoints.down("xs")]: {
                width: "100vw"
            }
        },
        drawerClosed: {
            width: "calc(100vw - 72px)",
            transition: theme.transitions.create("width", {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.leavingScreen
            }),
            [theme.breakpoints.down("xs")]: {
                width: "100vw"
            }
        },
        grid: {
            backgroundColor: "#fff"
        },
        paper: {
            padding: theme.spacing(2),
            textAlign: "center",
            color: theme.palette.text.secondary,
            backgroundColor: "rgba(0, 33, 69, 0.05)",
            height: "100%"
        },
        vizPaper: {
            padding: theme.spacing(2),
            height: "100%"
        },
        primaryStatisticContainer: {
            display: "flex",
            flexDirection: "column"
        },
        primaryStatisticValue: {
            overflow: "hidden",
            textOverflow: "ellipsis",
            fontSize: "calc((100vw - 240px)/16)"
        },
        primaryStatisticLabel: {
            paddingTop: "16px"
        },
        dateTimeContainer: {},
        textField: {
            marginLeft: theme.spacing(1),
            marginRight: theme.spacing(1),
            width: 200
        },
        dateField: {
            width: 200,
            marginRight: theme.spacing(1),
            marginTop: 0,
            marginBottom: 0
        },
        dateTimeButton: {
            marginLeft: "8px",
            marginTop: "6px"
        }
    })
);
