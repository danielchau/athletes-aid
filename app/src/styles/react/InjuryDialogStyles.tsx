import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

export const injuryDialogStyles = makeStyles((theme: Theme) =>
    createStyles({
        appBar: {
            position: "relative"
        },
        title: {
            marginLeft: theme.spacing(2),
            flex: 1
        },
        dialogTitle: {
            padding: "0"
        },
        dialogPaper: {
            height: "100%",
            maxWidth: "600px",
            width: "50%",
            minWidth: "200px",
            position: "fixed",
            bottom: "0",
            right: "0",
            margin: "0"
        },
        dialogContentContainer: {
            padding: "0"
        },
        dialogContent: {
            padding: "16px"
        },
        dialogContentPaper: {
            backgroundColor: "#fafafa",
            padding: "8px",
            marginBottom: "8px"
        },
        notePaper: {
            backgroundColor: "#ffffff",
            padding: "8px",
            marginBottom: "8px"
        },
        noteContainer: {
            display: "flex",
            flexDirection: "row"
        },
        noteIcon: {
            margin: "auto",
            marginLeft: "8px",
            marginRight: "16px"
        },
        newNoteContainer: {
            display: "flex",
            flexDirection: "row",
            width: "100%",
            justifyContent: "center",
            alignItems: "center",
            alignSelf: "center"
        },
        newNote: {
            flexGrow: 1,
            paddingRight: "8px"
        },
        sendButton: {
            color: "#F2A71E"
        },
        activeSwitchContainer: {
            display: "flex",
            flexDirection: "row",
            justifyContent: "center",
            alignContent: "center",
            alignSelf: "center",
            paddingBottom: "8px"
        }
    })
);
