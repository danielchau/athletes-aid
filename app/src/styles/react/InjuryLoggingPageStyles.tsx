import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

export const injuryLoggingPageStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            padding: "8px"
        },
        backButton: {
            marginRight: theme.spacing(1)
        },
        completedContainer: {
            height: "calc(100% - 116px)",
            textAlign: "center",
            paddingTop: "20%"
        },
        instructions: {
            marginTop: theme.spacing(1),
            marginBottom: theme.spacing(1)
        },
        paper: {
            color: theme.palette.text.secondary,
            backgroundColor: "#fafafa"
        },
        paperContent: {
            height: "100%",
            padding: "16px",
            paddingRight: "8px",
            overflow: "auto"
        },
        stepper: {
            backgroundColor: "#fafafa",
            borderRadius: "4px"
        },
        loggingContent: {
            height: "calc(100% - 168px)",
            marginBottom: "52px",
            marginTop: "8px",
            overflow: "hidden"
        },
        loggingBottomButtons: {
            position: "fixed",
            bottom: 0,
            right: 0,
            height: "52px",
            padding: "8px"
        },
        multilineInput: {
            width: "100%"
        }
    })
);
