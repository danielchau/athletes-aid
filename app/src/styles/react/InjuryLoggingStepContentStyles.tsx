import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

export const injuryLoggingStepContentStyles = makeStyles((theme: Theme) =>
    createStyles({
        formControl: {
            minWidth: 200,
            width: "50%",
            paddingRight: "8px",
            paddingBottom: "16px"
        },
        textInput: {
            minWidth: 200,
            width: "100%",
            paddingRight: "8px",
            paddingBottom: "16px"
        },
        multilineInput: {
            width: "100%",
            paddingBottom: "16px",
            margin: "0px"
        },
        dateInput: {
            minWidth: 200,
            width: "50%",
            margin: "0px",
            paddingRight: "8px",
            paddingBottom: "16px"
        },
        checkboxInput: {
            minWidth: 200,
            width: "50%",
            margin: "0px",
            paddingRight: "8px",
            paddingBottom: "23px",
            paddingTop: "7px"
        },
        instructions: {
            textAlign: "center",
            marginBottom: theme.spacing(3),
            color: theme.palette.text.secondary,
            width: "100%"
        }
    })
);
