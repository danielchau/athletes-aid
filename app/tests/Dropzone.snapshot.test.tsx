import * as React from 'react'
import Dropzone from '../src/components/Dropzone'
import renderer from 'react-test-renderer'

it('Dropzone renders correctly', () => {
  const tree = renderer.create(<Dropzone setNewAthletes={false} />).toJSON()
  expect(tree).toMatchSnapshot()
})
