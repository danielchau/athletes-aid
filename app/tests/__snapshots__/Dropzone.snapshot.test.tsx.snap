// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropzone renders correctly 1`] = `
<div
  onBlur={[Function]}
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "border": "1px dashed #F2A71E",
      "display": "flex",
      "flexDirection": "column",
      "flexGrow": 1,
      "height": "calc(100% - 16px)",
      "justifyContent": "center",
      "margin": "8px",
      "textAlign": "center",
    }
  }
  tabIndex={0}
>
  <input
    autoComplete="off"
    multiple={true}
    onChange={[Function]}
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    tabIndex={-1}
    type="file"
  />
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root"
    focusable="false"
    role="presentation"
    style={
      Object {
        "height": "2em",
        "width": "2em",
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM14 13v4h-4v-4H7l5-5 5 5h-3z"
    />
  </svg>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    Drag 'n' drop some files here, or click to select files
  </p>
</div>
`;
